File: app/src/main/java/cn/ifafu/ifafu/mvp/score/ScoreContract.java
package cn.ifafu.ifafu.mvp.score;

import java.util.List;
import java.util.Map;

import cn.ifafu.ifafu.data.entity.Response;
import cn.ifafu.ifafu.data.entity.Score;
import cn.ifafu.ifafu.mvp.base.i.IView;
import cn.ifafu.ifafu.mvp.base.i.IZFModel;
import cn.ifafu.ifafu.mvp.base.i.IZFPresenter;
import io.reactivex.Observable;

class ScoreContract {

    interface View extends IView {
        void setYearTermOptions(int option1, int option2);

        void setRvScoreData(List<Score> data);

        void setYearTermData(List<String> years, List<String> terms);

        void setYearTermTitle(String year, String term);

        void setIESText(String big, String little);

        void setCntText(String big, String little);

        void setGPAText(String text);
    }

    interface Presenter extends IZFPresenter {

        void updateFromNet();

        void switchYearTerm(int op1, int op2);

        void openFilterActivity();

        void updateIES();
    }

    interface Model extends IZFModel {

        Observable<Response<List<Score>>> getScoresFromNet(String year, String term);

        Observable<Response<List<Score>>> getScoresFromNet();

        List<Score> getScoresFromDB(String year, String term);

        Observable<Map<String, List<String>>> getYearTermList();

        Observable<Map<String, String>> getYearTerm();

        void save(List<Score> list);

        void delete(String year, String term);

        void deleteAllOnlineCourse();
    }

}

File: app/src/main/java/cn/ifafu/ifafu/mvp/score/ScoreContract.kt
package cn.ifafu.ifafu.mvp.score

import cn.ifafu.ifafu.data.entity.Score
import cn.ifafu.ifafu.data.entity.YearTerm
import cn.ifafu.ifafu.mvp.base.i.IView
import cn.ifafu.ifafu.mvp.base.i.IZFModel
import cn.ifafu.ifafu.mvp.base.i.IZFPresenter
import io.reactivex.Observable

class ScoreContract {
    interface View : IView {
        fun showIESDetail(text: String)
        fun setYearTermOptions(option1: Int, option2: Int)
        fun setRvScoreData(data: List<Score>)
        fun setYearTermData(years: List<String>, terms: List<String>)
        fun setYearTermTitle(year: String, term: String)
        fun setIESText(big: String, little: String)
        fun setCntText(big: String, little: String)
        fun setGPAText(text: String)
    }

    interface Presenter : IZFPresenter {
        fun updateFromNet()
        fun switchYearTerm(op1: Int, op2: Int)
        fun openFilterActivity()
        fun updateIES()
        fun checkIESDetail()
    }

    interface Model : IZFModel {
        /**
         * 获取成绩并保存
         * @param year 学年
         * @param term 学期
         * @return 成绩
         */
        fun getScoresFromNet(year: String, term: String): Observable<List<Score>>

        fun getScoresFromNet(): Observable<MutableList<Score>>

        fun getScoresFromDB(year: String, term: String): List<Score>

        fun getYearTermList(): Observable<YearTerm>

        fun getYearTerm(): Observable<Map<String, String>>
    }
}